### PR TITLE
Added Prompt

### PR DESCRIPTION
**Pull request by VoidNulll**
Added Prompt.js, a class that works as Prompt from the Klasa handler for djs,
Example command for prompt: wave

### CODE REVIEW

File: example/src/modules/Core/commands/Wave.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                        return this.sendMessage(msg.channel, 'I can\'t wave to someone who does not exist.');
+                    }
+                }
+                const username = (user.guild && user.nick) || (user.guild && `${user.username}#${user.discriminator}`) || `${user.username}#${user.discriminator}`;
+                return this.sendMessage(msg.channel, `*Waves to ${username}*`);
+            } catch (err) {
+                if (err === 'Prompt timed out' || err === 'Invalid usage found. Prompt ended.') {
+                    return;
+                }
+                return this.sendMessage(msg.channel, `Error: \`\`\`js\n${err}\`\`\``);
```
Comment (Khaazz) [inline example/src/modules/Core/commands/Wave.js]: I would use this.sendError() here for consistency

File: example/src/modules/Core/commands/Wave.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            const prompt = new Prompt(this.axon, msg.author.id, msg.channel, {
+                timeout: 10000,
+            });
+            try {
+                const returnedMes = await prompt.run('Who would you like to me to wave to?', {
+                    deleteTimeoutMess: 10000,
+                    timeoutMess: 'Well, I guess you don\'t want to wave at anyone then.',
+                });
+                console.log(returnedMes.content);
+                let user = Resolver.member(msg.channel.guild, returnedMes.content) || Resolver.user(this.axon.client, returnedMes.content);
```
Comment (Khaazz) [inline example/src/modules/Core/commands/Wave.js]: importing Resolver works but I suggest just calling this.Resolver

File: example/src/modules/Core/commands/Wave.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                    return;
+                }
+                return this.sendMessage(msg.channel, `Error: \`\`\`js\n${err}\`\`\``);
+            }
+        } else {
+            try {
+                let user = Resolver.member(msg.channel.guild, args.join(' ')) || Resolver.user(this.axon.client, args.join(' '));
+                if (!user) {
+                    if (Util.id.test(args[0])) {
+                        user = await this.axon.client.getRESTUser(args[0]);
```
Comment (Khaazz) [inline example/src/modules/Core/commands/Wave.js]: That works, but you should use `this.bot` here.

File: src/Utility/Prompt.js [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            }, timeoutTime);
+        });
+    }
+
+    /**
+     * Checker for this._onMsgCreate
+     *
+     * @param {Object} msg The message object to check against.
+     */
+    async _checker(msg) {
```
Comment (Khaazz) [inline src/Utility/Prompt.js]: This method should probably be refactored a bit so checker only returns a boolean for instance and al the processing happens in the onMessageCreate methot.

**Comment by VoidNulll** [comment]
Updated.

**Comment by Khaazz** [comment]
Merging this as it is complete and functional since the last bug we noticed earlier was corrected as well.

**Pull request merged**